---
title: "Gene-level DE analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(sleuth)
library(tidyverse)

```

```{r sleuth}
samples <- tibble(id = dir(file.path("mondet/kallisto_mite"))) %>% mutate(path = paste0("mondet/kallisto_mite/", id)) %>%
  left_join(read_tsv("mondet/treatments.tsv", col_types =  cols())) %>% mutate(sample = id, condition = shortName)
```

#transcript to gene table
```{r}
t2g = read.csv("/Users/lazzataibekova/Dropbox (OIST)/varroa microbiome/genebank_gene_map.csv", header=T)
```

### Comparing males vs young
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("young", "males")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionyoung')
results_table = sleuth_results(so, 'conditionyoung')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
###Young vs Phoretic

```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("young", "phoretic")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionyoung')
results_table = sleuth_results(so, 'conditionyoung')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Phoretics vs caged
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("phoretic", "phoreticCaged")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionphoreticCaged')
results_table = sleuth_results(so, 'conditionphoreticCaged')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Phoretic vs arresting
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("phoretic", "arresting")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionphoretic')
results_table = sleuth_results(so, 'conditionphoretic')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### –êrresting vs. pre-laying
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("arresting", "prelaying")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionprelaying')
results_table = sleuth_results(so, 'conditionprelaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Pre-laying vs. Laying
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("prelaying", "laying")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionprelaying')
results_table = sleuth_results(so, 'conditionprelaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Laying vs. post-laying
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("laying", "postlaying")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionpostlaying')
results_table = sleuth_results(so, 'conditionpostlaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Post-laying vs. Emerging
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("postlaying", "emerging")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionpostlaying')
results_table = sleuth_results(so, 'conditionpostlaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Post-laying vs. non-laying
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("postlaying", "nonlaying")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionpostlaying')
results_table = sleuth_results(so, 'conditionpostlaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Laying vs. non-laying
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("laying", "nonlaying")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionnonlaying')
results_table = sleuth_results(so, 'conditionnonlaying')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Emerging vs. Phoretic
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("emerging", "phoretic")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionphoretic')
results_table = sleuth_results(so, 'conditionphoretic')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```
### Emerging vs. Young
```{r}
so <- sleuth_prep(dplyr::filter(samples, shortName %in% c("emerging", "young")), target_mapping = t2g,aggregation_column = 'ens_gene', gene_mode = T)
```

```{r}
plot_pca(so, color_by = 'condition')
```

```{r}
so <- sleuth_fit(so, ~condition, 'full')
so <- sleuth_fit(so, ~1, 'reduced')
so <- sleuth_lrt(so, 'reduced', 'full')
```

```{r}
sleuth_table <- sleuth_results(so, 'reduced:full', 'lrt')
sleuth_significant <- dplyr::filter(sleuth_table, qval <= 0.05)
dim(sleuth_significant)
head(sleuth_significant)
```

```{r}
models(so)
```
Number of Differentially expressed genes (TRUE):
```{r}
so = sleuth_wt(so, 'conditionyoung')
results_table = sleuth_results(so, 'conditionyoung')
Number_DE_genes = table(results_table$qval <= 0.05)
Number_DE_genes
```
```{r}
DE_genes = subset(results_table, qval <= 0.05)
DE_genes
up_down_regulated = table(DE_genes$b > 0)
up_down_regulated
```